experimental/list_formatter/src/list_formatter.rs

```diff
@@ -50,23 +50,25 @@ impl ListFormatter {
         &'c self,
         values: &[&str],
         empty: fn() -> B,
```

Comment:
Suggestion: Merge `empty` and `single`:

```rust
create_output_buffer(&str, usize)
```

For `empty`, you can set the `usize` argument to `0`